include/ppx/bitmap.h

```diff
@@ -195,6 +196,28 @@ class Bitmap
     std::vector<char> mInternalStorage = {};
 };
 
+template <typename PixelDataType>
+void Bitmap::Fill(PixelDataType r, PixelDataType g, PixelDataType b, PixelDataType a)
+{
+    PPX_ASSERT_MSG(mData != nullptr, "data is null");
+    PPX_ASSERT_MSG(mFormat != Bitmap::FORMAT_UNDEFINED, "format is undefined");
+
+    PixelDataType rgba[4] = {r, g, b, a};
+
+    const uint32_t channelCount = Bitmap::ChannelCount(mFormat);
+
+    char* pData = mData;
+    for (uint32_t y = 0; y < mHeight; ++y) {
+        for (uint32_t x = 0; x < mWidth; ++x) {
+            PixelDataType* pPixelData = reinterpret_cast<PixelDataType*>(pData);
+            for (uint32_t c = 0; c < channelCount; ++c) {
```

Comment:
memcpy?

Reply:
I'll leave the for just in this case because I think it's a little more readable and I bet the compiler optimizes to the same thing.